integration/src/main/java/scripts/testscripts/ControlledBigQueryDatasetLifecycle.java

```diff
@@ -117,7 +115,10 @@ protected void doUserJourney(TestUserSpecification testUser, WorkspaceApi worksp
     String tableName = table.getTableId().getTable();
 
     // Workspace reader can read the table
-    var readTable = readerBqClient.getTable(table.getTableId());
+    // This is the reader's first use of cloud APIs after being added to the workspace, so we
+    // retry this operation until cloud IAM has properly synced.
+    var readTable = ClientTestUtils.getWithRetryOnException(() ->
+        readerBqClient.getTable(table.getTableId()), 20, Duration.ofSeconds(30));
```

Comment:
i wonder if we shouldn't use an exponential backoff, assuming the majority of cases happen within the first 15-30 seconds.

Reply:
Two thoughts here:
 1. All of the calls to the method use the same parameters. Is it worth parameterizing? Maybe just put the values into the `getWithRetryOnException` utility. If it turns out we need different parameters sometime later, we can factor a general method.
 2. Exponential would be incrementally faster. Since this is for test, it doesn't matter a lot. My rough calculations says  if we usually succeed in under 15 seconds, then this will make a regression run take a minute longer.

Reply:
sgtm, inlined the parameters into the method. Exponential backoff might be a bit cleaner, but I shortened the poll delay to 15s so this shouldn't add too much additional delay